src/main/java/com/uipath/uipathpackage/Utility.java

```diff
@@ -162,12 +63,26 @@ private File getTempDir() throws IOException {
         return tempDir;
     }
 
-    private void extractResourcesToTempFolder(File tempDir, File jarfile, TaskListener listener) throws IOException {
+    /**
+     * Validates the param for null or empty check
+     *
+     * @param param Param to validate
+     * @param s     Error Message
+     */
+    public void validateParams(String param, String s) {
```

Comment:
Some methods in this class are static while others are not. 
If you think all these are stateless (which they should be since they are utility methods), then you could make the entire class static.

Reply:
Ideally yes, entire class should be static, however I had to make it nonstatic as well as some method as nonstatic to have it tested. I am using mockito which can't test static methods, so to have proper testing I need to make the main caller method and other method which is being called by them as non static.